### ISSUE TITLE
gradient descent backpropagation

### ISSUE THREAD

**Issue by RInterested** [RInterested/statistics#44]
![backpropagation](https://cloud.githubusercontent.com/assets/9312897/15807170/a0303b44-2b24-11e6-974b-bbbae9390f1b.png)
